src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -1,38 +1,12 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
-/**
- * Parameterize this service and add its implementation in a separate class.
- */
-public interface MachineService {
+public interface MachineService<T extends Machine> {
+    List<T> getAll(Class<? extends Machine> type);
```

Comment:
Consider creating a local variable in your implementation of `getAll(Class type)` method. It can improve readability and make the code cleaner.